### ISSUE TITLE
Allow importing of records without a DOI

### ISSUE THREAD

**Issue by michael** [substance/texture#483]
Currently we only allow importing of bibliographic records that have a DOI. 

See: https://github.com/substance/texture/pull/480#discussion_r181580524

To add support for entries without DOI's we'd need to make sure we don't end up with a lot of duplicates. Some options:

- Be okay with duplicates being created for entries without a DOI
- Implement a checksum over JSON object so we can check if an entry has exactly the same content and if yes skip it.

**Comment by Melissa37** [comment]
Hi there

Why is this so important? Is it because you are still trying to build a database of references? :-)

Many journals do not have DOIs yet, and some journals with DOIs did not update their archives and add DOIs to them. Some books have DOIs, but not all.

We have to allow for all reference types to not have DOIs.

eLife has schematron that detects duplicate references I think @JGilbert-eLife can you explain how we do it?

Cheers
M

**Comment by oliver----** [comment]
Yeah 😀 no, it isn't because of that. As you said there will be a lot of references without DOI. Also we are allowing to Import from reference Managers, such as Zotero. We are just concerned to introduce problems by allowing to import the same record twice.

**Comment by Melissa37** [comment]
Cool! I hope James might be able to share how we do the check via Schematron if that helps?

**Comment by robsalasco** [comment]
any news about this feature? I'm using texture for a spanish journal but it's near impossible to have all the doi codes

**Comment by oliver----** [comment]
> Are you planning on including schematron in texture or did you just want to know the logic we were following so you could replicate it in something else?

When integrating Texture into XPub we would like create an interface between Texture and Schematron. In addition to that, we are adding a simple validator into Texture that runs checks inside the editor on-the-fly.

**Comment by eternoinvisible** [comment]
> Yeah  no, it isn't because of that. As you said there will be a lot of references without DOI. Also we are allowing to Import from reference Managers, such as Zotero. We are just concerned to introduce problems by allowing to import the same record twice.

I tried to import the csl jason (generated with zotero) file of citations without DOI and it didn´t allow me to do it because of lack of DOI error
